### DISCUSSION TITLE
Auto-format images to webp if the request Accept: image/webp header is found

### DISCUSSION THREAD

**Discussion by docmattman** [directus/directus#15918] [Feature Requests]
Since next-gen image formats are important for Core Web Vitals scores and there is lots of support for webp image format, it'd be great if there was a way to auto-format images to webp when possible.  Browses (like Chrome) send the `Accept` request header, like this:

```
Accept: image/webp,image/apng,image/svg+xml,image/*,*/*;q=0.8
```

I know we can use the `format=webp` param when making an image asset request.  But if an older browser tries to load an image that has the webp format specifically applied, it won't work.  So ideally the image asset service would be able to detect if webp is supported (by checking the `Accept` request header for `image/webp`) and serve the webp format for that request if found, or fallback to the original if not.

Perhaps there could be some sort of `AUTO_FORMAT_IMAGES` env var that could be set to allow this detection to be enabled and done automatically.

**Comment by azrikahar** [comment]
That'd be a great improvement indeed! Coincidentally it was also suggested by #15796 and #13626, so do give #13626 an upvote so that we can consolidate the votes which make it easier to gauge the overall community interest 👍

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
